### PR TITLE
Version comparison

### PR DESCRIPTION
**Pull request by emymaria**
Compare two different versions of algorithm

https://user-images.githubusercontent.com/68675647/131247315-f252612b-9204-4425-b1f1-dcaa4120c555.mp4

### CODE REVIEW

**Comment by salmma** [comment]
solves #149

File: src/app/components/algorithms/dialogs/compare-version-dialog.component.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    'property',
+    'currentVersionValue',
+    'compareVersionValue',
+  ];
+  tableColumns: string[] = [
+    'Properties',
+    'Current Version',
+    'Compared Version',
+  ];
+  comparisons: ComparedData[];
```
Comment (lharzenetter) [inline src/app/components/algorithms/dialogs/compare-version-dialog.component.ts:31]: I guess `differences` is the better choice of words.

**Review by lharzenetter — CHANGES_REQUESTED** [review]

**Review by salmma — CHANGES_REQUESTED** [review]
See my comments.
Please also do the same procedure for implementations, as there are also revisions available :)

**Review by lharzenetter — APPROVED** [review]

**Review by salmma — CHANGES_REQUESTED** [review]
As already mentioned, please also implement the revision comparison for implementations

**Review by salmma — CHANGES_REQUESTED** [review]
* The "Compare Versions" button should be removed at other tabs than the "General" tab, as no comparison is made, e.g., between previous related algorithms and the latest list - so this could confuse the user.
* same for Implementation
Bildschirmfoto 2021-09-13 um 18 07 45

* the "Compare Versions" buttons differ between Algorithm and Implementation view

**Review by salmma — APPROVED** [review]

**Pull request merged**